src/window/building/depot.c

```diff
@@ -0,0 +1,441 @@
+{
+    int y_offset = window_building_get_vertical_offset(c, 28);
+    c->help_id = 0;
+    outer_panel_draw(c->x_offset, y_offset, 29, 28);
+    text_draw_centered(title, c->x_offset, y_offset + 10, 16 * c->width_blocks, FONT_LARGE_BLACK, 0);
+    inner_panel_draw(c->x_offset + 16, y_offset + 42, c->width_blocks - 2, 21);
+}
+
+void window_building_draw_depot_select_source(building_info_context *c)
+{
+    draw_order_source_destination_background(c, translation_for(TR_BUILDING_INFO_DEPOT_SELECT_SOURCE_TITLE));
+}
+
+void draw_depot_select_source_destination(building_info_context *c) {
+    int y_offset = window_building_get_vertical_offset(c, 28);
+
+    scrollbar.x = c->x_offset + 16 * (c->width_blocks - 2) - 26;
+    scrollbar.y = y_offset + 46;
+    scrollbar_draw(&scrollbar);
+
+    int index = 0, offset = scrollbar.scroll_position;
+
+    for (building *b = building_first_of_type(BUILDING_GRANARY); b; b = b->next_of_type) {
+        if (index >= MAX_VISIBLE_ROWS) {
+            break;
+        }
+        if (building_storage_resource_max_storable(b, data.target_resource_id) > 0) {
+            if (offset > 0) {
+                offset--;
+            } else {
+                button_border_draw(c->x_offset + 18, y_offset + 46 + ROW_HEIGHT * index,
+                    16 * (c->width_blocks - 2) - 4 - (scrollbar.max_scroll_position > 0 ? 39 : 0),
+                    22, data.storage_building_focus_button_id == index + 1);
+                text_draw_label_and_number_centered(lang_get_string(28, b->type),
+                    b->storage_id, "", c->x_offset + 32, y_offset + 52 + ROW_HEIGHT * index,
+                    16 * (c->width_blocks - 2) - 4 - (scrollbar.max_scroll_position > 0 ? 39 : 0),
+                    FONT_SMALL_PLAIN, COLOR_WHITE);
+                index++;
+            }
+        }
+    }
+    for (building *b = building_first_of_type(BUILDING_WAREHOUSE); b; b = b->next_of_type) {
+        if (index >= MAX_VISIBLE_ROWS) {
+            break;
+        }
+        if (building_storage_resource_max_storable(b, data.target_resource_id) > 0) {
+            if (offset > 0) {
+                offset--;
+            } else {
+                button_border_draw(c->x_offset + 18, y_offset + 46 + ROW_HEIGHT * index,
+                    16 * (c->width_blocks - 2) - 4 - (scrollbar.max_scroll_position > 0 ? 39 : 0),
+                    22, data.storage_building_focus_button_id == index + 1);
+                text_draw_label_and_number_centered(lang_get_string(28, b->type),
+                    b->storage_id, "", c->x_offset + 32, y_offset + 52 + ROW_HEIGHT * index,
+                    16 * (c->width_blocks - 2) - 4 - (scrollbar.max_scroll_position > 0 ? 39 : 0),
+                    FONT_SMALL_PLAIN, COLOR_WHITE);
+                index++;
+            }
+        }
+    }
+}
+
+void window_building_draw_depot_select_source_foreground(building_info_context *c)
+{
+    draw_depot_select_source_destination(c);
```

Comment:
Why have this function call the other one?

You could always paste the contents of `draw_depot_select_source_destination()` here and get rid of that function instead.

Reply:
Done.